### PR TITLE
asrv: split off database handling.

### PR DESCRIPTION
**Pull request by koriakin**

### CODE REVIEW

File: python/veles/asrv/srv.py [hunk 1]
```diff
@@ -198,31 +134,14 @@ def remove_conn(self, conn):
...
-            """, (raw_obj_id, key, self.packer.pack(val)))
+            self.db.set_data(node.id, key, val)
         for key, val in bindata.items():
-            c.execute("""
-                INSERT INTO object_bindata (obj_id, name, data)
-                VALUES (?, ?, ?)
-            """, (raw_obj_id, key, val))
-        self.db.commit()
+            self.db.set_bindata(node.id, key, 0, val, False)
         obj = self.get(obj_id)
```
Comment (altran01) [inline python/veles/asrv/srv.py]: "fix" this?

File: python/veles/asrv/srv.py [hunk 2]
```diff
@@ -198,31 +134,14 @@ def remove_conn(self, conn):
...
-        """, (raw_obj_id, raw_parent, pos[0], pos[1]))
-        for tag in tags:
-            c.execute("""
-                INSERT INTO object_tag (obj_id, name) VALUES (?, ?)
-            """, (raw_obj_id, tag))
-        for key, val in attr.items():
-            c.execute("""
-                INSERT INTO object_attr (obj_id, name, data) VALUES (?, ?, ?)
-            """, (raw_obj_id, key, self.packer.pack(val)))
+        node = Node(id=obj_id, parent=parent, pos_start=pos[0],
```
Comment (altran01) [inline python/veles/asrv/srv.py]: Shouldn't it be added to self.objs after db.create?

Comment (koriakin) [inline python/veles/asrv/srv.py]: This is going to be fixed in another PR

File: python/veles/db.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+    def set_pos(self, obj_id, pos_start, pos_end):
+        if not isinstance(obj_id, NodeID):
+            raise TypeError('node id has wrong type')
+        if (not isinstance(pos_start, six.integer_types)
+                and pos_start is not None):
+            raise TypeError('pos_start has to be an int')
+        if (not isinstance(pos_end, six.integer_types)
+                and pos_end is not None):
+            raise TypeError('pos_start has to be an int')
```
Comment (altran01) [inline python/veles/db.py]: s/start/end

File: python/veles/tests/db/test_db.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+        self.assertEqual(n2.parent, node.parent)
+        self.assertEqual(n2.tags, node.tags)
+        self.assertEqual(n2.attr, node.attr)
+        self.assertEqual(n2.pos_start, node.pos_start)
+        self.assertEqual(n2.pos_end, node.pos_end)
+        self.assertEqual(n2.data, set())
+        self.assertEqual(n2.bindata, {})
+        db2.close()
+        os.unlink(path)
+        os.rmdir(d)
```
Comment (altran01) [inline python/veles/tests/db/test_db.py]: On windows files in temp never get deleted automatically, so I'd change it so that we try to clean-up before failing the test to avoid unnecessary additional garbage.

File: python/veles/tests/db/test_db.py [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+                         -0x123456789abcdef123456789abcdef)
+        db.set_attr(node.id, 'int', None)
+        db.set_attr(node.id, 'bool', True)
+        db.set_attr(node.id, 'meh', 'meh')
+        n3 = db.get(node.id)
+        self.assertEqual(set(n3.attr.keys()), {
+            'meh', 'bool', 'bytes', 'str', 'list', 'dict', 'id', 'bindata',
+            'long', 'neglong',
+        })
+        self.assertNotIn('int', n3.attr)
```
Comment (altran01) [inline python/veles/tests/db/test_db.py]: Isn't this a bit redundant with previous assert?
